sim/midas/src/main/scala/midas/core/FPGATop.scala

```diff
@@ -79,6 +79,25 @@ case class HostMemChannelParams(
     idBits   = idBits)
 }
 
+/** Specifies the AXI4 configuration passed on to the driver. */
+case class AXI4Config(
```

Comment:
Until there's a need to serialize something more than just variable width fields, you should just use AXI4BundleParameters instead of introducing a new class.